tests/test_cpp/polars/TestWindowProcessor.cpp

```diff
@@ -401,22 +437,189 @@ TEST(Series, rolling_mean_triangle) {
+        Series({1.6666666666666667, 2.0, 2.333333333333333}, {1, 2, 3})
+    ) << "Expect " << " matching center = true";
+
+    EXPECT_PRED2(
+        Series::almost_equal,
+        Series({1, 2, 3}, {1, 2, 3}).rolling(5, polars::Mean(), 1, false, false,
+                                             polars::WindowProcessor::WindowType::triang),
+        Series({1.0, 1.3333333333333333, 1.6666666666666667}, {1, 2, 3})
+    ) << "Expect " << " matching center = false";
+
+    EXPECT_PRED2(
+        Series::almost_equal,
+        Series({1, 2, 3}, {1, 2, 3}).rolling(4, polars::Mean(), 1, true, false,
+                                             polars::WindowProcessor::WindowType::triang),
+        Series({1.25, 1.7142857142857142, 2.2857142857142856}, {1, 2, 3})
+    ) << "Expect " << " matching center = true";
+
+    EXPECT_PRED2(
+        Series::almost_equal,
+        Series({1, 2, 3}, {1, 2, 3}).rolling(4, polars::Mean(), 1, false, false,
+                                             polars::WindowProcessor::WindowType::triang),
+        Series({1.0, 1.25, 1.7142857142857142}, {1, 2, 3})
+    ) << "Expect " << " matching center = false";
+
+    // If we are just one above window we need to add NAN at each side for window centred
+
+    EXPECT_PRED2(
+        Series::almost_equal,
+        Series({1, 2, 3, 4}, {1, 2, 3, 4}).rolling(8, 1, true, false,
+                                                   polars::WindowProcessor::WindowType::triang).mean(),
+        Series({1.875, 2.1818181818181817, 2.5, 2.8181818181818183}, {1, 2, 3, 4})
     ) << "Expect " << "no NANs because min periods is 1.";
+
 }
 
+
+TEST(Series, rolling_mean_triangle__larger_window) {
+
+    arma::vec input_values = {0.1, 0.3, 0.5, 0.4, 0.7, 0.9, 0.3, 0.1};
+    arma::vec input_timestamps = {1, 2, 3, 4, 5, 6, 7, 8};
+
+    Series input = Series(input_values, input_timestamps);
+
+    int window_size = 20;
+    // THESE ARE NOT CENTER = TRUE values :P
+    arma::vec expected_values_center_false = {0.10000000000000002, 0.14999999999999999, 0.21111111111111114, 0.25624999999999998,
+                                 0.29599999999999999, 0.34166666666666667, 0.37551020408163266,0.38906250000000003};
+
+    auto actual = input.rolling(window_size, polars::Mean(), 1, false, false, polars::WindowProcessor::WindowType::triang);
+
+    EXPECT_PRED2(Series::almost_equal, actual, Series(expected_values_center_false, input_timestamps));
+
+    arma::vec expected_values_center_true = {0.39687500000000003, 0.40454545454545465, 0.4175, 0.4293650793650794,
+                                              0.43984375, 0.44682539682539685, 0.44250000000000006, 0.4318181818181819};
+
+    actual = input.rolling(window_size, polars::Mean(), 1, true, false, polars::WindowProcessor::WindowType::triang);
+
+    EXPECT_PRED2(Series::almost_equal, actual, Series(expected_values_center_true, input_timestamps));
+
+}
+
+// NEED TO FINISH OFF THIS TEST.
```

Comment:
Should this be marked with 'todo' - or is it part of this PR to finish it?

Reply:
Part of the PR. There are still some outstanding cases.